Course/Course/Presentation/Subviews/CourseHeaderView.swift

```diff
@@ -20,38 +20,54 @@ struct CourseHeaderView: View {
     @Binding private var isAnimatingForTap: Bool
     @Environment(\.isHorizontal) private var isHorizontal
     
-    private let collapsedHorizontalHeight: CGFloat = 230
-    private let collapsedVerticalHeight: CGFloat = 260
-    private let expandedHeight: CGFloat = 300
+    private var collapsedHorizontalHeight: CGFloat {
+        186 + (!viewModel.shouldHideMenuBar ? 44 : 0)
+    }
+    private var collapsedVerticalHeight: CGFloat {
+        216 + (!viewModel.shouldHideMenuBar ? 44 : 0)
+    }
+
+    private var expandedHeight: CGFloat {
+        220 + (!viewModel.shouldHideMenuBar ? 80 : 0) + (viewModel.shouldShowUpgradeButton ? 42+20 : 0)
+    }
```

Comment:
I think, we should elaborate these hardcoded integer constants. It may create confusions later.

Reply:
I guess this isn't required.